tt_eager/tt_dnn/op_library/eltwise_binary/eltwise_binary_op.cpp

```diff
@@ -160,23 +166,49 @@ void EltwiseBinary::validate(const std::vector<Tensor>& input_tensors) const {
             TT_FATAL(input_tensor_a.memory_config().memory_layout == input_tensor_b.memory_config().memory_layout);
             TT_FATAL(input_tensor_a.shard_spec().value() == input_tensor_b.shard_spec().value());
         }
-        if (this->output_mem_config.is_sharded()) {
+        if(!output_tensors.empty() && output_tensors.at(0).has_value()){
+            const auto& out_tensor = output_tensors.at(0).value();
+            if (out_tensor.memory_config().is_sharded()) {
+                TT_FATAL(input_tensor_a.memory_config().memory_layout == out_tensor.memory_config().memory_layout);
+            } else {
+                TT_FATAL(out_tensor.memory_config().memory_layout == TensorMemoryLayout::INTERLEAVED);
+            }
+        }
+        else if (this->output_mem_config.is_sharded()) {
```

Comment:
I think the code would be much simpler if you just did the check for which mem config to check once.

`auto out_mem_config = (!output_tensors.empty() && output_tensors.at(0).has_value()) ? output_tensors.at(0).value().memory_config() : this->output_mem_config;`

And then use out_mem_config in these checks, instead of adding these duplicated checks everywhere

Reply:
Thanks. I have made the changes.